jax/experimental/sparse/bcoo.py

```diff
@@ -1033,9 +1032,8 @@ def _bcoo_dot_general_batch_rule(batched_args, batch_dims, *, dimension_numbers,
 ad.defjvp(bcoo_dot_general_p, _bcoo_dot_general_jvp_lhs, None, _bcoo_dot_general_jvp_rhs)
 ad.primitive_transposes[bcoo_dot_general_p] = _bcoo_dot_general_transpose
 batching.primitive_batchers[bcoo_dot_general_p] = _bcoo_dot_general_batch_rule
-
-mlir.register_lowering(
-    bcoo_dot_general_p, _bcoo_dot_general_default_lowering)
+bcoo_dot_general_p.def_impl(partial(dispatch.apply_primitive, bcoo_dot_general_p))
```

Comment:
We have a helper function for this these days called [`dispatch.simple_impl`](https://github.com/google/jax/blob/612a94016098ae148e79f261049d1f91b24b6660/jax/_src/dispatch.py#L127), if you'd like to use it. We don't use it in too many places yet, and either way is OK.

Reply:
Great idea! Done